### PR TITLE
SEQNG-664: Parallelize keyword collection

### PR DESCRIPTION
**Pull request by cquiroz**
Looking at the code I can tell we can run some of the keyword related tasks in parallel. After a serious fight with cats implicits I managed to get it working

I tested these against GPI without any issues. I hope the DHS can also accept this

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/SeqexecFailure.scala [hunk 1]
```diff
@@ -44,6 +46,8 @@ object SeqexecFailure {
   /** XMLRPC error while communicating with the GDS */
   final case class GDSXMLError(msg: String, url: Uri) extends SeqexecFailure
 
+  final case class SeqexecCombinedFailure(a: SeqexecFailure, b: SeqexecFailure) extends SeqexecFailure
```
Comment (tpolecat) [inline modules/seqexec/server/src/main/scala/seqexec/server/SeqexecFailure.scala]: Another solution in cases like this is to use `NonEmptyList[SeqexecFailure]` which gives you a free semigroup, so you don't need the reified constructor for `|+|`. Just FYI. But this is fine.

**Comment by jluhrs** [comment]
Unfortunately, Science requested to preserve the order of the keywords. DHS puts the keywords in the order it receives them. That means we cannot send the different groups in parallel :(

**Comment by jluhrs** [comment]
We could collect the keywords in parallel though.

**Comment by tpolecat** [comment]
> Science requested to preserve the order of the keywords

Weird. Any idea why?

**Comment by cquiroz** [comment]
Would it make sense to make it "backend" optional?

**Pull request closed without merging**